back-end/objects/RImageFolder.py

```diff
@@ -248,23 +248,34 @@ def __init__(
         self._init_next_records()
 
     def add_records(self, records: List[Tuple[str, int]], correct: bool):
-        buffer = self.buffer_correct if correct else self.buffer_incorrect
         next_record = self.next_correct if correct else self.next_incorrect
-
         paths = [to_unix(record[0]) for record in records]
 
         # 1. update db with the result
-        db_update_many_by_paths(self.db_conn, self.table_name, paths, ('classified',),
-                                [(correct,) for _ in paths])
-
-        # 2. update buffer
-        buffer.extend(records)
-
-        # 3. update next record datastructure
+        if correct:
+            value_list = [(self.CLS_CORRECT,) for _ in paths]
+        else:
+            value_list = [(self.CLS_INCORRECT,) for _ in paths]
+        db_update_many_by_paths(self.db_conn, self.table_name, paths, ('classified',), value_list)
+
+        # 2. check duplicates
+        to_remove = set(records)
+        self.buffer_correct = [item for item in self.buffer_correct]
+        self.buffer_incorrect = [item for item in self.buffer_incorrect if item[0] not in to_remove]
+
+        # 3. update buffer
+        buffer = self.buffer_correct if correct else self.buffer_incorrect
+        if buffer == []:
+            buffer.extend(records)
+        else:
+            dic = dict(buffer)
+            dic.update(dict(records))
+            buffer = list(dic.items())
```

Comment:
I think there is still some problem with this implementation. The reason why it seems to work is because line 273 `buffer = list(dic.items())` is a local assignment and does not change `self.buffer_correct` and `self.buffer_incorrect`. 

I think the easiest way to solve this is to update the record entirely. Please take a look at my latest commit to your branch.